### PR TITLE
[DOC][REF] K-Shape rename and add references

### LINKED ISSUE #1595: [DOC] k-Shape missing reference, and should be k-shape not k-shapes
### Describe the issue linked to the documentation

we should have a reference in the docstring of kshape to [1]
Also, the original name of the model is kshape, but we have kshape`s`

[1] Paparrizos, John, and Luis Gravano. "k-shape: Efficient and accurate clustering of time series." Proceedings of the 2015 ACM SIGMOD international conference on management of data. 2015.

### Suggest a potential alternative/fix

_No response_

### PR DESCRIPTION
**Pull request by Cyril-Meyer**
#### Reference Issues/PRs

Fixes #1595

#### What does this implement/fix? Explain your changes.

Replace `TimeSeriesKShapes` with `TimeSeriesKShape` as the original name is K-Shape

#### Does your contribution introduce a new dependency? If yes, which one?

No.

#### Any other comments?

The citation format is ACM instead of proposed in #1595 as it's the one proposed by the official website and also include DOI.
I'm not sure about the PR title tags, its [DOC] for the reference, but I don't know how to classify the renaming between [REF] and [MNT].

##### For all contributions
- [x] I've added myself to the [list of contributors](https://github.com/aeon-toolkit/aeon/blob/main/.all-contributorsrc). Alternatively, you can use the [@all-contributors](https://allcontributors.org/docs/en/bot/usage) bot to do this for you.
- [x] The PR title starts with either [ENH], [MNT], [DOC], [BUG], [REF], [DEP] or [GOV] indicating whether the PR topic is related to enhancement, maintenance, documentation, bugs, refactoring, deprecation or governance.

### CODE REVIEW

**Review by hadifawaz1999 — CHANGES_REQUESTED** [review]
Thanks for taking care of this, however it requires deprecation policy, https://www.aeon-toolkit.org/en/stable/developer_guide/deprecation.html

**Comment by Cyril-Meyer** [comment]
> Thanks for taking care of this, however it requires deprecation policy, https://www.aeon-toolkit.org/en/stable/developer_guide/deprecation.html

Thanks for the info, I had completely forgotten about that, even though I had already seen it for other modules I worked on.
For now, I've deprecated the original class as I previously see in `networks/_cnn.py` for v0.12.0.
* Kept the `TimeSeriesKShapes` version with new `TimeSeriesKShape` in module `__init__` and documentation
* Only use `TimeSeriesKShape` For tests and base module, we

**Review by hadifawaz1999 — APPROVED** [review]
LGTM ! @TonyBagnall  any thoughts for the deprecation version ? 0.12 or 1.0 ?

File: aeon/clustering/_k_shapes.py [hunk 1]
```diff
@@ -3,12 +3,197 @@
...
+        }
+
+    def _score(self, X, y=None):
+        return np.abs(self.inertia_)
+
+
+# TODO: remove in v0.12.0
+@deprecated(
+    version="0.10.0",
+    reason="TimeSeriesKShapes class has been renamed to TimeSeriesKShape and "
```
Comment (TonyBagnall) [inline aeon/clustering/_k_shapes.py:193]: Maybe reword "TimeSeriesKShapes class has been renamed to TimeSeriesKShape.  The TimeSeriesKShapes version will be removed in version 1.0.0"

**Comment by MatthewMiddlehurst** [comment]
Can rename any incorrect versions later, but I agree 1.0.0 is safer

File: aeon/clustering/_k_shapes.py (hunk 1 above)
Comment (Cyril-Meyer) [reply aeon/clustering/_k_shapes.py]: Done in 93635a123df1cc3ddc205e5d6d8d6f86ccc59012

**Comment by Cyril-Meyer** [comment]
Related to the previous work, I propose to rename the file k_shapes to k_shape (done in ed5041b960b1b165f4f7dbb59a14c9def5db7e45).
Is there something to do about the deprecation ?

**Comment by MatthewMiddlehurst** [comment]
No deprecation need there, the public access is through `aeon.clustering` and that has not changed.

**Review by MatthewMiddlehurst — DISMISSED** [review]

**Comment by MatthewMiddlehurst** [comment]
LGTM unless @TonyBagnall has any more comments

**Review by hadifawaz1999 — DISMISSED** [review]
LGTM! @TonyBagnall all good for this one ?

**Comment by Cyril-Meyer** [comment]
> soz, been away. lgtm, but there is a conflict with main @Cyril-Meyer

This should be resolved.
Conflict came from #1944 from @hadifawaz1999 on `aeon/clustering/deep_learning/base.py` and its addition make that my changes on the file becamed useless :)

**Comment by hadifawaz1999** [comment]
> > soz, been away. lgtm, but there is a conflict with main @Cyril-Meyer
> 
> This should be resolved. Conflict came from #1944 from @hadifawaz1999 on `aeon/clustering/deep_learning/base.py` and its addition make that my changes on the file becamed useless :)

ah yes forgot we worked on the same file my bad !

**Review by hadifawaz1999 — APPROVED** [review]
